susemanager-utils/susemanager-sls/src/modules/kiwi_info.py

```diff
@@ -37,23 +39,54 @@ def parse_buildinfo(dest):
                 group[match.group('name')] = match.group('val')
     return ret
 
-# fallback for SLES11 Kiwi that does not create the buildinfo file
+# fallback for SLES11 Kiwi and for Kiwi NG that does not create the buildinfo file
 def guess_buildinfo(dest):
     ret = {'main': {}}
     files = __salt__['file.readdir'](dest)
 
     pattern_basename = re.compile(r"^(?P<basename>.*)\.packages$")
-    pattern_pxe = re.compile(r"^initrd-netboot")
+    pattern_pxe_initrd = re.compile(r"^initrd-netboot.*")
+    pattern_pxe_kiwi_ng_initrd = re.compile(r".*\.initrd\..*")
+    pattern_pxe_kernel = re.compile(r".*\.kernel\..*")
+    pattern_pxe_kiwi_ng_kernel = re.compile(r".*\.kernel$")
+    have_kernel = False
+    have_initrd = False
+
     for f in files:
         match = pattern_basename.match(f)
         if match:
             ret['main']['image.basename'] = match.group('basename')
 
-        match = pattern_pxe.match(f)
+        match = pattern_pxe_initrd.match(f) or pattern_pxe_kiwi_ng_initrd.match(f)
+        if match:
+            have_initrd = True
+
+        match = pattern_pxe_kernel.match(f) or pattern_pxe_kiwi_ng_kernel.match(f)
         if match:
-            ret['main']['image.type'] = 'pxe'
+            have_kernel = True
+
+    if have_kernel and have_initrd:
+        ret['main']['image.type'] = 'pxe'
+    return ret
+
+# Kiwi NG
+def parse_kiwi_result(dest):
+    path = os.path.join(dest, 'kiwi.result')
+    ret = {}
+    if __salt__['file.file_exists'](path):
+        try:
+            with open(path, 'rb') as f:
+                result = pickle.load(f)
+                ret['arch'] = result.xml_state.host_architecture
+                ret['basename'] = result.xml_state.xml_data.name
+                ret['type'] = result.xml_state.build_type.image
+                ret['filesystem'] = result.xml_state.build_type.filesystem
+                ret['initrd_system'] = result.xml_state.build_type.initrd_system
+        except:
+            log.exception("Loading kiwi.result")
```

Comment:
Is it ok to handle the exception and then return empty dict? Does it make sense to rethrow the exception after logging?

Reply:
If it fails, some info will be missing, but the image itself is usable, so I think we can continue here. The empty dict is handled  later.

Reply:
Please add a comment stating that the image will still be usable, otherwise, it is not really clear from the context